Add orator 0.9.9

aaronmussig opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in 
a comment on the PR to get the attention of the `staged-recipes` team. You can 
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/orator```) and found some lint.

Here's what I've got...


For **recipes/orator**:

 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.
 * There are 2 too many lines.  There should be one empty line at the end of the file.

aaronmussig commented:
Hi, if anyone can assist with this error I would greatly appreciate it. I'm unable to resolve the following error:

```
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Attempting to finalize metadata for orator
INFO:conda_build.metadata:Attempting to finalize metadata for orator
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
BUILD START: ['orator-0.9.9-py_0.tar.bz2']
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place


The following NEW packages will be INSTALLED:

    _libgcc_mutex:      0.1-conda_forge              conda-forge
    _openmp_mutex:      4.5-0_gnu                    conda-forge
    attrs:              19.3.0-py_0                  conda-forge
    ca-certificates:    2019.11.28-hecc5488_0        conda-forge
    cachecontrol:       0.12.6-py_0                  conda-forge
    cachy:              0.3.0-py_0                   conda-forge
    certifi:            2019.11.28-py36h9f0ad1d_1    conda-forge
    cffi:               1.14.0-py36hd463f26_0        conda-forge
    chardet:            3.0.4-py36_1003              conda-forge
    cleo:               0.7.6-py_0                   conda-forge
    clikit:             0.4.2-py_0                   conda-forge
    cryptography:       2.8-py36h72c5cf5_1           conda-forge
    dbus:               1.13.6-he372182_0            conda-forge
    entrypoints:        0.3-py36h9f0ad1d_1001        conda-forge
    expat:              2.2.9-he1b5a44_2             conda-forge
    gettext:            0.19.8.1-hc5be6a0_1002       conda-forge
    glib:               2.58.3-py36h6f030ca_1002     conda-forge
    html5lib:           1.0.1-py_0                   conda-forge
    idna:               2.9-py_1                     conda-forge
    importlib-metadata: 1.1.3-py36_0                 conda-forge
    importlib_metadata: 1.1.3-0                      conda-forge
    jeepney:            0.4.3-py_0                   conda-forge
    jsonschema:         3.2.0-py36_0                 conda-forge
    keyring:            18.0.1-py36_0                conda-forge
    ld_impl_linux-64:   2.33.1-h53a641e_8            conda-forge
    libffi:             3.2.1-he1b5a44_1006          conda-forge
    libgcc-ng:          9.2.0-h24d8f2e_2             conda-forge
    libgomp:            9.2.0-h24d8f2e_2             conda-forge
    libiconv:           1.15-h516909a_1005           conda-forge
    libstdcxx-ng:       9.2.0-hdf63c60_2             conda-forge
    lockfile:           0.12.2-py_1                  conda-forge
    msgpack-python:     1.0.0-py36hc9558a2_0         conda-forge
    ncurses:            6.1-hf484d3e_1002            conda-forge
    openssl:            1.1.1d-h516909a_0            conda-forge
    pastel:             0.2.0-py_0                   conda-forge
    pcre:               8.44-he1b5a44_0              conda-forge
    pexpect:            4.8.0-py36_0                 conda-forge
    pip:                20.0.2-py_2                  conda-forge
    pkginfo:            1.5.0.1-py_0                 conda-forge
    poetry:             1.0.5-py36_0                 conda-forge
    ptyprocess:         0.6.0-py_1001                conda-forge
    pycparser:          2.20-py_0                    conda-forge
    pylev:              1.3.0-py_0                   conda-forge
    pyopenssl:          19.1.0-py_1                  conda-forge
    pyparsing:          2.4.6-py_0                   conda-forge
    pyrsistent:         0.14.11-py36h516909a_1       conda-forge
    pysocks:            1.7.1-py36_0                 conda-forge
    python:             3.6.10-h9d8adfe_1009_cpython conda-forge
    python_abi:         3.6-1_cp36m                  conda-forge
    readline:           8.0-hf8c457e_0               conda-forge
    requests:           2.23.0-py36_0                conda-forge
    requests-toolbelt:  0.8.0-py_1                   conda-forge
    secretstorage:      3.1.2-py36_0                 conda-forge
    setuptools:         46.0.0-py36h9f0ad1d_2        conda-forge
    shellingham:        1.3.2-py_0                   conda-forge
    six:                1.14.0-py_1                  conda-forge
    sqlite:             3.30.1-hcee41ef_0            conda-forge
    tk:                 8.6.10-hed695b0_0            conda-forge
    tomlkit:            0.5.11-py36_0                conda-forge
    urllib3:            1.25.7-py36_0                conda-forge
    webencodings:       0.5.1-py_1                   conda-forge
    wheel:              0.34.2-py_1                  conda-forge
    xz:                 5.2.4-h14c3975_1001          conda-forge
    zipp:               3.1.0-py_0                   conda-forge
    zlib:               1.2.11-h516909a_1006         conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
WARNING: symlink_conda() is deprecated.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
WARNING: symlink_conda() is deprecated.
Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
INFO:conda_build.source:Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
INFO conda_build.source:download_to_cache(43): Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
INFO:conda_build.source:Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
INFO conda_build.source:download_to_cache(63): Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
Extracting download
source tree in: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work
export PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place
export BUILD_PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_build_env
export SRC_DIR=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work
Non-user install because user site-packages disabled
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-wlkqi707
Created temporary directory: /tmp/pip-req-tracker-uts6o8ny
Initialized build tracking at /tmp/pip-req-tracker-uts6o8ny
Created build tracker: /tmp/pip-req-tracker-uts6o8ny
Entered build tracker: /tmp/pip-req-tracker-uts6o8ny
Created temporary directory: /tmp/pip-install-n990gc66
Processing $SRC_DIR
  Created temporary directory: /tmp/pip-req-build-b3_rdcs6
  Added file://$SRC_DIR to build tracker '/tmp/pip-req-tracker-uts6o8ny'
    Created temporary directory: /tmp/pip-modern-metadata-gv4qeig4
    Preparing wheel metadata: started
    Running command /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpe02rf8tb
    Preparing wheel metadata: finished with status 'done'
  Source in /tmp/pip-req-build-b3_rdcs6 has version 0.9.9, which satisfies requirement orator==0.9.9 from file://$SRC_DIR
  Removed orator==0.9.9 from file://$SRC_DIR from build tracker '/tmp/pip-req-tracker-uts6o8ny'
Building wheels for collected packages: orator
  Created temporary directory: /tmp/pip-wheel-4hk0xv_e
  Destination directory: /tmp/pip-wheel-4hk0xv_e
  Building wheel for orator (PEP 517): started
  Running command /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmphp54t6sx
  Building wheel for orator (PEP 517): finished with status 'done'
  Created wheel for orator: filename=orator-0.9.9-py2.py3-none-any.whl size=172924 sha256=cf61b9b039199a16b70a45f34d8c7868976aa668da9d46bdb5e1fb7c33468f43
  Stored in directory: /tmp/pip-ephem-wheel-cache-wlkqi707/wheels/52/5d/71/f0451cb35ae4228315c375fd9cfa38d514e3a80593b594c586
Successfully built orator
Installing collected packages: orator
  Created temporary directory: /tmp/pip-unpacked-wheel-wllu9ffw

  changing mode of $PREFIX/bin/orator to 775
Successfully installed orator-0.9.9
Cleaning up...
  Removing source in /tmp/pip-req-build-b3_rdcs6
Removed build tracker: '/tmp/pip-req-tracker-uts6o8ny'

Resource usage statistics from building orator:
   Process count: 1
   CPU time: Sys=0:00:00.0, User=0:00:00.0
   Memory: 3.3M
   Disk usage: 1.2K
   Time elapsed: 0:00:02.1

Packaging orator
INFO:conda_build.build:Packaging orator
INFO conda_build.build:build(1556): Packaging orator
Packaging orator-0.9.9-py_0
INFO:conda_build.build:Packaging orator-0.9.9-py_0
INFO conda_build.build:bundle_conda(842): Packaging orator-0.9.9-py_0
number of files: 149
nm: dynamic_annotations.o: no symbols
WARNING (orator): run-exports library package conda-forge::python-3.6.10-h9d8adfe_1009_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Fixing permissions
Packaged license file.
Compressing to /tmp/tmp7cidfjgd/orator-0.9.9-py_0.tar.bz2
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
TEST START: /home/uqamussi_local/miniconda3/conda-bld/noarch/orator-0.9.9-py_0.tar.bz2
Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
INFO conda_build.variants:_combine_spec_dictionaries(188): Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
Renaming work directory,  /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work  to  /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work_moved_orator-0.9.9-py_0_noarch
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.

## Package Plan ##

  environment location: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl


The following NEW packages will be INSTALLED:

    _libgcc_mutex:     0.1-conda_forge              conda-forge
    _openmp_mutex:     4.5-0_gnu                    conda-forge
    backpack:          0.1-py36_1000                conda-forge
    blinker:           1.4-py_1                     conda-forge
    ca-certificates:   2019.11.28-hecc5488_0        conda-forge
    certifi:           2019.11.28-py36h9f0ad1d_1    conda-forge
    faker:             4.0.1-py_0                   conda-forge
    inflection:        0.3.1-py_1                   conda-forge
    krb5:              1.16.4-h2fd8d38_0            conda-forge
    lazy-object-proxy: 1.4.3-py36h516909a_1         conda-forge
    ld_impl_linux-64:  2.33.1-h53a641e_8            conda-forge
    libedit:           3.1.20170329-hf8c457e_1001   conda-forge
    libffi:            3.2.1-he1b5a44_1006          conda-forge
    libgcc-ng:         9.2.0-h24d8f2e_2             conda-forge
    libgomp:           9.2.0-h24d8f2e_2             conda-forge
    libpq:             12.2-hae5116b_0              conda-forge
    libstdcxx-ng:      9.2.0-hdf63c60_2             conda-forge
    mysql-connector-c: 6.1.11-hc9bbbdd_1005         conda-forge
    mysqlclient:       1.4.6-py36he1b5a44_0         conda-forge
    ncurses:           6.1-hf484d3e_1002            conda-forge
    openssl:           1.1.1d-h516909a_0            conda-forge
    orator:            0.9.9-py_0                   local      
    pendulum:          2.1.0-py36_0                 conda-forge
    pip:               20.0.2-py_2                  conda-forge
    psycopg2:          2.8.4-py36h72c5cf5_1         conda-forge
    python:            3.6.10-h9d8adfe_1009_cpython conda-forge
    python-dateutil:   2.8.1-py_0                   conda-forge
    python_abi:        3.6-1_cp36m                  conda-forge
    pytzdata:          2019.3-py_0                  conda-forge
    readline:          8.0-hf8c457e_0               conda-forge
    setuptools:        46.0.0-py36h9f0ad1d_2        conda-forge
    simplejson:        3.17.0-py36h516909a_0        conda-forge
    six:               1.14.0-py_1                  conda-forge
    sqlite:            3.30.1-hcee41ef_0            conda-forge
    text-unidecode:    1.3-py_0                     conda-forge
    tk:                8.6.10-hed695b0_0            conda-forge
    wheel:             0.34.2-py_1                  conda-forge
    wrapt:             1.12.1-py36h8c4c3a4_1        conda-forge
    xz:                5.2.4-h14c3975_1001          conda-forge
    zlib:              1.2.11-h516909a_1006         conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
WARNING: symlink_conda() is deprecated.
export PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl
export SRC_DIR=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/test_tmp
Traceback (most recent call last):
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/test_tmp/run_test.py", line 2, in <module>
    import orator
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/__init__.py", line 6, in <module>
import: 'orator'
    from .database_manager import DatabaseManager
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/database_manager.py", line 6, in <module>
    from .connectors.connection_factory import ConnectionFactory
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/connectors/__init__.py", line 4, in <module>
    from .mysql_connector import MySQLConnector
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/connectors/mysql_connector.py", line 4, in <module>
    from pendulum import Pendulum, Date
ImportError: cannot import name 'Pendulum'
Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
WARNING:conda_build.build:Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
WARNING conda_build.build:tests_failed(2201): Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
TESTS FAILED: orator-0.9.9-py_0.tar.bz2
```

aaronmussig commented:
Hi, @conda-forge/help-python if anyone can assist with this error I would greatly appreciate it. I'm unable to resolve the following error:

```
No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Attempting to finalize metadata for orator
INFO:conda_build.metadata:Attempting to finalize metadata for orator
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
BUILD START: ['orator-0.9.9-py_0.tar.bz2']
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place


The following NEW packages will be INSTALLED:

    _libgcc_mutex:      0.1-conda_forge              conda-forge
    _openmp_mutex:      4.5-0_gnu                    conda-forge
    attrs:              19.3.0-py_0                  conda-forge
    ca-certificates:    2019.11.28-hecc5488_0        conda-forge
    cachecontrol:       0.12.6-py_0                  conda-forge
    cachy:              0.3.0-py_0                   conda-forge
    certifi:            2019.11.28-py36h9f0ad1d_1    conda-forge
    cffi:               1.14.0-py36hd463f26_0        conda-forge
    chardet:            3.0.4-py36_1003              conda-forge
    cleo:               0.7.6-py_0                   conda-forge
    clikit:             0.4.2-py_0                   conda-forge
    cryptography:       2.8-py36h72c5cf5_1           conda-forge
    dbus:               1.13.6-he372182_0            conda-forge
    entrypoints:        0.3-py36h9f0ad1d_1001        conda-forge
    expat:              2.2.9-he1b5a44_2             conda-forge
    gettext:            0.19.8.1-hc5be6a0_1002       conda-forge
    glib:               2.58.3-py36h6f030ca_1002     conda-forge
    html5lib:           1.0.1-py_0                   conda-forge
    idna:               2.9-py_1                     conda-forge
    importlib-metadata: 1.1.3-py36_0                 conda-forge
    importlib_metadata: 1.1.3-0                      conda-forge
    jeepney:            0.4.3-py_0                   conda-forge
    jsonschema:         3.2.0-py36_0                 conda-forge
    keyring:            18.0.1-py36_0                conda-forge
    ld_impl_linux-64:   2.33.1-h53a641e_8            conda-forge
    libffi:             3.2.1-he1b5a44_1006          conda-forge
    libgcc-ng:          9.2.0-h24d8f2e_2             conda-forge
    libgomp:            9.2.0-h24d8f2e_2             conda-forge
    libiconv:           1.15-h516909a_1005           conda-forge
    libstdcxx-ng:       9.2.0-hdf63c60_2             conda-forge
    lockfile:           0.12.2-py_1                  conda-forge
    msgpack-python:     1.0.0-py36hc9558a2_0         conda-forge
    ncurses:            6.1-hf484d3e_1002            conda-forge
    openssl:            1.1.1d-h516909a_0            conda-forge
    pastel:             0.2.0-py_0                   conda-forge
    pcre:               8.44-he1b5a44_0              conda-forge
    pexpect:            4.8.0-py36_0                 conda-forge
    pip:                20.0.2-py_2                  conda-forge
    pkginfo:            1.5.0.1-py_0                 conda-forge
    poetry:             1.0.5-py36_0                 conda-forge
    ptyprocess:         0.6.0-py_1001                conda-forge
    pycparser:          2.20-py_0                    conda-forge
    pylev:              1.3.0-py_0                   conda-forge
    pyopenssl:          19.1.0-py_1                  conda-forge
    pyparsing:          2.4.6-py_0                   conda-forge
    pyrsistent:         0.14.11-py36h516909a_1       conda-forge
    pysocks:            1.7.1-py36_0                 conda-forge
    python:             3.6.10-h9d8adfe_1009_cpython conda-forge
    python_abi:         3.6-1_cp36m                  conda-forge
    readline:           8.0-hf8c457e_0               conda-forge
    requests:           2.23.0-py36_0                conda-forge
    requests-toolbelt:  0.8.0-py_1                   conda-forge
    secretstorage:      3.1.2-py36_0                 conda-forge
    setuptools:         46.0.0-py36h9f0ad1d_2        conda-forge
    shellingham:        1.3.2-py_0                   conda-forge
    six:                1.14.0-py_1                  conda-forge
    sqlite:             3.30.1-hcee41ef_0            conda-forge
    tk:                 8.6.10-hed695b0_0            conda-forge
    tomlkit:            0.5.11-py36_0                conda-forge
    urllib3:            1.25.7-py36_0                conda-forge
    webencodings:       0.5.1-py_1                   conda-forge
    wheel:              0.34.2-py_1                  conda-forge
    xz:                 5.2.4-h14c3975_1001          conda-forge
    zipp:               3.1.0-py_0                   conda-forge
    zlib:               1.2.11-h516909a_1006         conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
WARNING: symlink_conda() is deprecated.
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
WARNING: symlink_conda() is deprecated.
Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
INFO:conda_build.source:Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
INFO conda_build.source:download_to_cache(43): Source cache directory is: /home/uqamussi_local/miniconda3/conda-bld/src_cache
Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
INFO:conda_build.source:Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
INFO conda_build.source:download_to_cache(63): Found source in cache: orator-0.9.9_6fe7830c40.tar.gz
Extracting download
source tree in: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work
export PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place
export BUILD_PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_build_env
export SRC_DIR=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work
Non-user install because user site-packages disabled
Ignoring indexes: https://pypi.org/simple
Created temporary directory: /tmp/pip-ephem-wheel-cache-wlkqi707
Created temporary directory: /tmp/pip-req-tracker-uts6o8ny
Initialized build tracking at /tmp/pip-req-tracker-uts6o8ny
Created build tracker: /tmp/pip-req-tracker-uts6o8ny
Entered build tracker: /tmp/pip-req-tracker-uts6o8ny
Created temporary directory: /tmp/pip-install-n990gc66
Processing $SRC_DIR
  Created temporary directory: /tmp/pip-req-build-b3_rdcs6
  Added file://$SRC_DIR to build tracker '/tmp/pip-req-tracker-uts6o8ny'
    Created temporary directory: /tmp/pip-modern-metadata-gv4qeig4
    Preparing wheel metadata: started
    Running command /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpe02rf8tb
    Preparing wheel metadata: finished with status 'done'
  Source in /tmp/pip-req-build-b3_rdcs6 has version 0.9.9, which satisfies requirement orator==0.9.9 from file://$SRC_DIR
  Removed orator==0.9.9 from file://$SRC_DIR from build tracker '/tmp/pip-req-tracker-uts6o8ny'
Building wheels for collected packages: orator
  Created temporary directory: /tmp/pip-wheel-4hk0xv_e
  Destination directory: /tmp/pip-wheel-4hk0xv_e
  Building wheel for orator (PEP 517): started
  Running command /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/python /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/lib/python3.6/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmphp54t6sx
  Building wheel for orator (PEP 517): finished with status 'done'
  Created wheel for orator: filename=orator-0.9.9-py2.py3-none-any.whl size=172924 sha256=cf61b9b039199a16b70a45f34d8c7868976aa668da9d46bdb5e1fb7c33468f43
  Stored in directory: /tmp/pip-ephem-wheel-cache-wlkqi707/wheels/52/5d/71/f0451cb35ae4228315c375fd9cfa38d514e3a80593b594c586
Successfully built orator
Installing collected packages: orator
  Created temporary directory: /tmp/pip-unpacked-wheel-wllu9ffw

  changing mode of $PREFIX/bin/orator to 775
Successfully installed orator-0.9.9
Cleaning up...
  Removing source in /tmp/pip-req-build-b3_rdcs6
Removed build tracker: '/tmp/pip-req-tracker-uts6o8ny'

Resource usage statistics from building orator:
   Process count: 1
   CPU time: Sys=0:00:00.0, User=0:00:00.0
   Memory: 3.3M
   Disk usage: 1.2K
   Time elapsed: 0:00:02.1

Packaging orator
INFO:conda_build.build:Packaging orator
INFO conda_build.build:build(1556): Packaging orator
Packaging orator-0.9.9-py_0
INFO:conda_build.build:Packaging orator-0.9.9-py_0
INFO conda_build.build:bundle_conda(842): Packaging orator-0.9.9-py_0
number of files: 149
nm: dynamic_annotations.o: no symbols
WARNING (orator): run-exports library package conda-forge::python-3.6.10-h9d8adfe_1009_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Fixing permissions
Packaged license file.
Compressing to /tmp/tmp7cidfjgd/orator-0.9.9-py_0.tar.bz2
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
TEST START: /home/uqamussi_local/miniconda3/conda-bld/noarch/orator-0.9.9-py_0.tar.bz2
Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
INFO conda_build.variants:_combine_spec_dictionaries(188): Adding in variants from /tmp/tmp54sq3xpb/info/recipe/conda_build_config.yaml
Renaming work directory,  /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work  to  /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/work_moved_orator-0.9.9-py_0_noarch
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
WARNING:conda.base.context:Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.
WARNING conda.base.context:use_only_tar_bz2(632): Conda is constrained to only using the old .tar.bz2 file format because you have conda-build installed, and it is <3.18.3.  Update or remove conda-build to get smaller downloads and faster extractions.

## Package Plan ##

  environment location: /home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl


The following NEW packages will be INSTALLED:

    _libgcc_mutex:     0.1-conda_forge              conda-forge
    _openmp_mutex:     4.5-0_gnu                    conda-forge
    backpack:          0.1-py36_1000                conda-forge
    blinker:           1.4-py_1                     conda-forge
    ca-certificates:   2019.11.28-hecc5488_0        conda-forge
    certifi:           2019.11.28-py36h9f0ad1d_1    conda-forge
    faker:             4.0.1-py_0                   conda-forge
    inflection:        0.3.1-py_1                   conda-forge
    krb5:              1.16.4-h2fd8d38_0            conda-forge
    lazy-object-proxy: 1.4.3-py36h516909a_1         conda-forge
    ld_impl_linux-64:  2.33.1-h53a641e_8            conda-forge
    libedit:           3.1.20170329-hf8c457e_1001   conda-forge
    libffi:            3.2.1-he1b5a44_1006          conda-forge
    libgcc-ng:         9.2.0-h24d8f2e_2             conda-forge
    libgomp:           9.2.0-h24d8f2e_2             conda-forge
    libpq:             12.2-hae5116b_0              conda-forge
    libstdcxx-ng:      9.2.0-hdf63c60_2             conda-forge
    mysql-connector-c: 6.1.11-hc9bbbdd_1005         conda-forge
    mysqlclient:       1.4.6-py36he1b5a44_0         conda-forge
    ncurses:           6.1-hf484d3e_1002            conda-forge
    openssl:           1.1.1d-h516909a_0            conda-forge
    orator:            0.9.9-py_0                   local      
    pendulum:          2.1.0-py36_0                 conda-forge
    pip:               20.0.2-py_2                  conda-forge
    psycopg2:          2.8.4-py36h72c5cf5_1         conda-forge
    python:            3.6.10-h9d8adfe_1009_cpython conda-forge
    python-dateutil:   2.8.1-py_0                   conda-forge
    python_abi:        3.6-1_cp36m                  conda-forge
    pytzdata:          2019.3-py_0                  conda-forge
    readline:          8.0-hf8c457e_0               conda-forge
    setuptools:        46.0.0-py36h9f0ad1d_2        conda-forge
    simplejson:        3.17.0-py36h516909a_0        conda-forge
    six:               1.14.0-py_1                  conda-forge
    sqlite:            3.30.1-hcee41ef_0            conda-forge
    text-unidecode:    1.3-py_0                     conda-forge
    tk:                8.6.10-hed695b0_0            conda-forge
    wheel:             0.34.2-py_1                  conda-forge
    wrapt:             1.12.1-py36h8c4c3a4_1        conda-forge
    xz:                5.2.4-h14c3975_1001          conda-forge
    zlib:              1.2.11-h516909a_1006         conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
WARNING: symlink_conda() is deprecated.
export PREFIX=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl
export SRC_DIR=/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/test_tmp
Traceback (most recent call last):
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/test_tmp/run_test.py", line 2, in <module>
    import orator
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/__init__.py", line 6, in <module>
import: 'orator'
    from .database_manager import DatabaseManager
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/database_manager.py", line 6, in <module>
    from .connectors.connection_factory import ConnectionFactory
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/connectors/__init__.py", line 4, in <module>
    from .mysql_connector import MySQLConnector
  File "/home/uqamussi_local/miniconda3/conda-bld/orator_1584078715058/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.6/site-packages/orator/connectors/mysql_connector.py", line 4, in <module>
    from pendulum import Pendulum, Date
ImportError: cannot import name 'Pendulum'
Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
WARNING:conda_build.build:Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
WARNING conda_build.build:tests_failed(2201): Tests failed for orator-0.9.9-py_0.tar.bz2 - moving package to /home/uqamussi_local/miniconda3/conda-bld/broken
TESTS FAILED: orator-0.9.9-py_0.tar.bz2
```

npavlovikj commented:
I think `from pendulum import Pendulum, Date` should probably be `from pendulum import pendulum, date`.

scopatz commented:
CC @conda-forge/help-python

npavlovikj commented on recipes/orator/meta.yaml:
```diff
@@ -0,0 +1,47 @@
+    - psycopg2
+    - mysqlclient
+    - backpack
+    - pendulum
```
Can you please try `pendulum <2` instead? This might be related, https://stackoverflow.com/questions/57616144/cant-create-exe-with-pycharm-as-it-complains-pendulum-locales.

aaronmussig replied:
Works well thanks, I brought the poetry versions across and it seems to work locally now.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/orator```) and found it was in an excellent condition.

npavlovikj commented on recipes/orator/meta.yaml:
```diff
@@ -0,0 +1,87 @@
+  home: https://orator-orm.com/
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.txt
```
Looks like the license file is part of the PyPI tarball, `orator-0.9.9/LICENSE`, so please use `license_file: LICENSE` here and remove the additional LICENSE.txt file.

aaronmussig replied:
You're right - that's been changed.

npavlovikj commented on recipes/orator/meta.yaml:
```diff
@@ -0,0 +1,87 @@
+
+requirements:
+  host:
+    - python =2.7|>=3.5,<3.7  # should be =2.7|>=3.5,<4 but backpack/pyaml clashes.
```
I think you can just use `python <3.7` here and in the `run` section.

aaronmussig replied:
That's true but in this case, it's just using what they explicitly support (with the exception of >=3.7,<4 due to the clash.

npavlovikj commented on recipes/orator/meta.yaml:
```diff
@@ -0,0 +1,87 @@
+  host:
+    - python =2.7|>=3.5,<3.7  # should be =2.7|>=3.5,<4 but backpack/pyaml clashes.
+    - pip
+    - poetry >=0.12a3
```
Is `poetry` needed for building the package?

aaronmussig replied:
I can't get it to build without it.

npavlovikj commented:
The recipe LGTM - thank you for contributing @aaronmussig 🎉
